DigitalLibraryofTennessee/DLTN_XSLT
# Add dateIssued values and reorder elements

**Jira Issue**: [DPLA-233](https://jirautk.atlassian.net/browse/DPLA-233)

* Other Relevant Links - [Order of XML Elements](https://dltn-technical-docs.readthedocs.io/en/latest/style/xsl.html)

## What does this Pull Request do?

This pull requests includes dateIssued in the transformation and places these values in dateCreated[@edtf] tags in the resulting metadata (lines 59-72). A sample file from Phoenix that has a dateIssued tag has also been included for testing. Because many of the student publications that were recently remediated have a dateIssued rather than a dateCreated tag, making these changes allows the metadata being shared with DPLA to be more complete.

Because changes were already being made, I also updated the order of the elements within the XSLT so that they follow the new order established on DLTN Technical Docs (link above).

## How should this be tested?

To test:

* Run the transform against the sample data (phoenix.xml)
* Run the transform against another UTK set of data that includes dateCreated values
* See if any blank or empty nodes are created.
* Check to see if the elements follow the ordering in https://dltn-technical-docs.readthedocs.io/en/latest/style/xsl.html

## Additional Notes:

I'm uncertain what level of edtf DPLA can handle. Phoenix has seasons included in the metadata that may or may not be helpful. If felt to be better, we could take the non-edtf dateIssued value but this would change things so that we have two different resulting dateCreated tags instead of the one dateCreated with encoding.

I'm hoping to get this set for the March 18th DPLA ingest. If we can get things ready, I will go ahead and reingest sets in Repox that will be affected by the dateIssued change. All sets can be reharvested if we'd like to get the new XML order in place.

## Interested parties

@markpbaggett @CanOfBees
